docs/ref/templates/language.txt

```diff
@@ -475,6 +475,150 @@ it also defines the content that fills the hole in the *parent*. If there were
+        <div class="card">
+            <h3>{{ title }}</h3>
+            <p>{{ content }}</p>
+        </div>
+    {% endpartialdef card %}
+
+Render the partial using the :ttag:`partial` tag:
+
+.. code-block:: html+django
+
+    {% partial card %}
+
+Fragment reuse
+--------------
+
+Once defined, you can reuse a partial multiple times:
+
+.. code-block:: html+django
+
+    {% partialdef user-info %}
+        <div class="user">
+            <span>{{ user.username }}</span>
+            <span>{{ user.email }}</span>
+        </div>
+    {% endpartialdef %}
+
+    {% block content %}
+        <h2>Authors</h2>
+        {% for user in authors %}
+            {% partial user-info %}
+        {% endfor %}
+
+        <h2>Editors</h2>
+        {% for user in editors %}
+            {% partial user-info %}
+        {% endfor %}
+    {% endblock %}
+
+The partial content will be rendered each time the named partial is used,
+with the current template context.
+
+Inline partials
+---------------
+
+Define and render a partial inline using the ``inline`` argument:
+
+.. code-block:: html+django
+
+    {% partialdef header inline %}
+        <header class="site-header">
+            <h1>{{ site_name }}</h1>
+        </header>
+    {% endpartialdef %}
+
+This defines the partial and renders it immediately in place.
+
+Direct partial access
+---------------------
+
+Access partials directly using the ``#`` syntax. In views, use the
+:func:`~django.shortcuts.render` shortcut:
+
+.. code-block:: python
+
+    # In views.py
+    from django.shortcuts import render
+
+
+    def product_card_partial(request, product_id):
+        product = get_object_or_404(Product, id=product_id)
+        return render(request, "products/list.html#product-card", {"product": product})
+
+This approach is useful for AJAX responses, HTMX requests, or when you need
+only a specific part of a template. Use partials with the :ttag:`include` tag:
+
+.. code-block:: html+django
+
+    {% include "products/list.html#product-card" %}
+
+Context handling
+----------------
+
+Template partials render with the current template context. They work as
+expected in loops and with context variables:
+
+.. code-block:: html+django
+
+    {% partialdef product-card %}
+        <div class="product" data-id="{{ product.id }}">
+            <h3>{{ product.name }}</h3>
+            <p>${{ product.price }}</p>
+        </div>
+    {% endpartialdef %}
+
+    {% for product in products %}
+        {% partial product-card %}
+    {% endfor %}
+
+If you need to adjust the context, use the :ttag:`with` tag:
+
+.. code-block:: html+django
+
+    {% with title="Special Offer" price=product.sale_price %}
+        {% partial product-card %}
+    {% endwith %}
+
+
+.. admonition:: Partial name uniqueness
+
+    You cannot define multiple partials with the same name in the same template.
+    If you do, the last partial definition will overwrite previous ones and be
+    the one that gets rendered.
```

Comment:
Is there a viable reason to do allow this? Should we error instead so that issues can be picked up during development? It feels like this would be something that people wouldn't want to silently happen.

Reply:
Now it raises a TemplateSyntaxError during parsing. 
```python
template_source = """
        {% partialdef duplicate %}DUPLICATE-CONTENT{% endpartialdef %}
        {% partialdef duplicate %}DUPLICATE-CONTENT-2{% endpartialdef %}
        """
with self.assertRaisesMessage(
    TemplateSyntaxError,
    "Partial 'duplicate' is already defined in the 'template.html' template.",
):
    Template(template_source, origin=Origin(name="template.html"))
```

Reply:
removed from the docs also.